velox/exec/RowsStreamingWindowBuild.cpp

```diff
@@ -101,13 +122,17 @@ void RowsStreamingWindowBuild::noMoreInput() {
 }
 
 std::shared_ptr<WindowPartition> RowsStreamingWindowBuild::nextPartition() {
-  VELOX_CHECK(hasNextPartition());
-  return windowPartitions_[++outputPartition_];
+  return outputPartition();
 }
 
 bool RowsStreamingWindowBuild::hasNextPartition() {
-  return !windowPartitions_.empty() &&
-      outputPartition_ + 2 <= windowPartitions_.size();
+  // Remove the processed output partition from the queue.
```

Comment:
hasNextPartition() should only be a query API. nextPartition() should be updating the windowPartitions_.

Reply:
@aditi-pandit @xiaoxmeng If we decide to remove the processed window partition in the nextPartition method, an exception may be thrown in certain cases. This is because hasNextPartition might return true, but when we actually retrieve the window partition in nextPartition, it could be empty after the removal. Therefore, we need to update windowPartition_ within the hasNextPartition method.